src/apps/competitions/models.py

```diff
@@ -6,23 +7,75 @@
 from utils.data import PathWrapper
 from utils.storage import BundleStorage
 
+logger = logging.getLogger()
+
 
 class Competition(models.Model):
     title = models.CharField(max_length=256)
     logo = models.ImageField(upload_to=PathWrapper('logos'), null=True, blank=True)
-    created_by = models.ForeignKey(settings.AUTH_USER_MODEL, null=True, blank=True, on_delete=models.SET_NULL, related_name="competitions")
+    created_by = models.ForeignKey(settings.AUTH_USER_MODEL, null=True, blank=True, on_delete=models.SET_NULL,
+                                   related_name="competitions")
     created_when = models.DateTimeField(default=now)
     collaborators = models.ManyToManyField(settings.AUTH_USER_MODEL, related_name="collaborations", blank=True)
     published = models.BooleanField(default=False)
     secret_key = models.UUIDField(default=uuid.uuid4, unique=True, null=True, blank=True)
+    is_migrating = models.BooleanField(default=False)
 
     def __str__(self):
-        return "competition-{0}-{1}".format(self.title, self.pk)
+        return f"competition-{self.title}-{self.pk}"
 
     @property
     def bundle_dataset(self):
         return CompetitionCreationTaskStatus.objects.get(resulting_competition=self).dataset
 
+    def apply_phase_migration(self, current_phase, next_phase):
+        '''
+        Does the actual migrating of submissions from current_phase to next_phase
+
+        :param current_phase: The phase object to transfer submissions from
+        :param next_phase: The new phase object we are entering
+        '''
+        logger.info(f"Checking for submissions that may still be running competition pk={self.pk}")
+
+        if not current_phase.submissions.filter(status='Scoring' or 'Cancelled' or 'Finished').exists():
```

Comment:
I think this is the correct logic for this query:
```python
if not current_phase.submissions.exclude(status__in=['Scoring', 'Cancelled', 'Finished']).exists()
```

Reply:
Used Scoring.CANCELLED, etc. and added NONE as well to prevent typos from causing problems. Corrected the Scoring status logger